### PR TITLE
Feature/home page

### PR DESCRIPTION
**Pull request by MirahSaqib**

### CODE REVIEW

File: app/controllers/hospitals_controller.rb [hunk 1]
```diff
@@ -1,9 +1,10 @@
 class HospitalsController < ApplicationController
   #POST /hospitals
   def list_hospitals
+    found = User.find_by(email: params[:email]).nil?
```
Comment (aliadnanaslam7) [inline app/controllers/hospitals_controller.rb]: blank?

File: app/controllers/hospitals_controller.rb [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
     respond_to do |format|
-      if User.find_by(email: params[:email]).nil?
-        format.html { redirect_to give_email_path, danger: t('pages.wrong_email') }
+      if found
```
Comment (aliadnanaslam7) [inline app/controllers/hospitals_controller.rb]: condition

File: app/controllers/pages_controller.rb [hunk 3]
```diff
@@ -1,10 +1,10 @@
 class PagesController < ApplicationController
-  layout 'static_pages_layout', only: [:about, :privacy_policy, :contact, :home]
+  layout 'static_pages_layout', only: [:about, :privacy_policy, :contact ]
 
   # GET /
   def home
     respond_to do |format|
-      format.html
+      format.html {render layout: false }
```
Comment (aliadnanaslam7) [inline app/controllers/pages_controller.rb]: space

File: app/controllers/passwords_controller.rb [hunk 4]
```diff
@@ -1,12 +1,18 @@
 class PasswordsController < Devise::PasswordsController
+  private :resource_params
+
   def create
-    user = User.find_by(email: params[:user][:email])
-    if user && user.hospital.subdomain != request.subdomain
+    user = User.find_by(email: params[:user][:email], hospital_id: @current_hospital.id)
+    if user.nil?
```
Comment (aliadnanaslam7) [inline app/controllers/passwords_controller.rb]: blank?

File: app/controllers/prescriptions_controller.rb [hunk 5]
```diff
@@ -65,7 +67,7 @@ def show
       format.pdf do
         render pdf: "Prescription No. #{@prescription.id}",
         page_size: 'A4',
-        template: "prescriptions/show.html.erb",
+        template: "prescriptions/pdf.html.erb",
```
Comment (aliadnanaslam7) [inline app/controllers/prescriptions_controller.rb]: format

File: app/controllers/sessions/sessions_controller.rb [hunk 6]
```diff
@@ -1,26 +1,28 @@
...
-
+  skip_before_filter :allow_params_authentication!
+  
   # POST /resource/sign_in
   def create
-    user = User.where(email: (params[:user][:email]), hospital_id: @current_hospital.id)
-    if !user.first.nil? && user.first.valid_password?(params[:user][:password])
+    user = User.find_by(email: (params[:user][:email]), hospital_id: @current_hospital.id)
+    if !user.nil? && user.valid_password?(params[:user][:password])
       if !session[:user_id].nil?
```
Comment (aliadnanaslam7) [inline app/controllers/sessions/sessions_controller.rb]: .present?

File: app/views/devise/sessions/new.html.erb [hunk 7]
```diff
@@ -1,41 +1,44 @@
...
-        </div>
-      <% end %>
+        <% if devise_mapping.rememberable? %>
+          <div class="form-row form-group justify-content-center">
+            <%= f.check_box :remember_me, class: 'remember-me-checkbox' %> &nbsp
+            <%= f.label :remember_me %>
+          </div>
+        <% end %>
+
+        <%- if devise_mapping.recoverable? && controller_name != 'passwords' && controller_name != 'registrations' %>
```
Comment (aliadnanaslam7) [inline app/views/devise/sessions/new.html.erb]: .include

File: app/views/medicines/_medicine_form.html.erb [hunk 8]
```diff
@@ -16,9 +16,9 @@
         <%= f.text_field :sale_price, class: 'form-control', required: true, type: 'number' %>
       </div>
 
-      <div class="col-sm-1"></div>
+      <div class="col-sm-2"></div>
 
-      <div class="col-sm-1"></div>
+        <%= f.hidden_field :quantity, value: 0 unless request.url.include?('edit') %>
```
Comment (aliadnanaslam7) [inline app/views/medicines/_medicine_form.html.erb]: persisted? / new_record?

File: app/views/pages/home.html.erb [hunk 9]
```diff
@@ -1,27 +1,138 @@
...
+              <div class="alert alert-success">
+            <% elsif key == "alert" %>
+              <div class="alert alert-danger">
+            <% else %>
+              <div class="alert alert-<%= key %>">
+            <% end %>
+            <%= value %>
+            <button type="button" class="close" data-dismiss="alert"> &times; </button>
+            </div>
+            <% flash.discard(key) %>
```
Comment (aliadnanaslam7) [inline app/views/pages/home.html.erb]: flash.now()

File: app/views/prescriptions/pdf.html.erb [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+              <th>Category</th>
+              <th>Sale Price</th>
+              <th>Quantity</th>
+            </tr>
+          </thead>
+          <tbody >
+            <% @prescription.prescription_medicines.each do |prescription_medicine| %>
+              <tr>
+                <th><%= prescription_medicine.id %></th>
+                <% if can? :read, prescription_medicine.medicine%>
```
Comment (aliadnanaslam7) [inline app/views/prescriptions/pdf.html.erb]: space

File: app/views/sale_orders/index.html.erb [hunk 11]
```diff
@@ -37,7 +37,9 @@
                   </td>
                   <td><%= order.created_at.strftime("%Y-%m-%d") %></td>
                   <td>$ <%= order.total_sale_price %></td>
-                  <td class="<%= order.status == SaleOrder::STATUS[:draft] ? 'text-danger' : 'text-success' %>" ><%= order.status.upcase %></td>
+                  <td class="text-center"><span class="<%= order.status == SaleOrder::STATUS[:draft] ? 'text-light rounded rounded-sm px-3 py-1 bg-danger' : 'text-light rounded rounded-sm px-2 py-1 bg-success' %>" ><%= order.status.downcase %></span>
```
Comment (aliadnanaslam7) [inline app/views/sale_orders/index.html.erb]: move helper

File: app/controllers/hospitals_controller.rb (hunk 1 above)
Comment (MirahSaqib) [reply app/controllers/hospitals_controller.rb]: done!

File: app/controllers/hospitals_controller.rb (hunk 2 above)
Comment (MirahSaqib) [reply app/controllers/hospitals_controller.rb]: done!

File: app/controllers/pages_controller.rb (hunk 3 above)
Comment (MirahSaqib) [reply app/controllers/pages_controller.rb]: done!

File: app/controllers/sessions/sessions_controller.rb (hunk 6 above)
Comment (MirahSaqib) [reply app/controllers/sessions/sessions_controller.rb]: done!

File: app/controllers/prescriptions_controller.rb (hunk 5 above)
Comment (MirahSaqib) [reply app/controllers/prescriptions_controller.rb]: done!

File: app/views/devise/sessions/new.html.erb (hunk 7 above)
Comment (MirahSaqib) [reply app/views/devise/sessions/new.html.erb]: done!

File: app/views/prescriptions/pdf.html.erb (hunk 10 above)
Comment (MirahSaqib) [reply app/views/prescriptions/pdf.html.erb]: done!

File: app/views/medicines/_medicine_form.html.erb (hunk 8 above)
Comment (MirahSaqib) [reply app/views/medicines/_medicine_form.html.erb]: done!

File: app/controllers/passwords_controller.rb (hunk 4 above)
Comment (MirahSaqib) [reply app/controllers/passwords_controller.rb]: done!

File: app/views/pages/home.html.erb (hunk 9 above)
Comment (MirahSaqib) [reply app/views/pages/home.html.erb]: done!

**Pull request merged**